src/saml.ts

```diff
@@ -1077,37 +1078,42 @@ class SAML {
           }
         }
 
-        subjectConfirmation = subject[0].SubjectConfirmation
-          ? subject[0].SubjectConfirmation[0]
-          : null;
-        confirmData =
-          subjectConfirmation && subjectConfirmation.SubjectConfirmationData
-            ? subjectConfirmation.SubjectConfirmationData[0]
-            : null;
-        if (subject[0].SubjectConfirmation && subject[0].SubjectConfirmation.length > 1) {
-          msg = "Unable to process multiple SubjectConfirmations in SAML assertion";
-          throw new Error(msg);
-        }
-
-        if (subjectConfirmation) {
-          if (confirmData && confirmData.$) {
-            const subjectNotBefore = confirmData.$.NotBefore;
-            const subjectNotOnOrAfter = confirmData.$.NotOnOrAfter;
-            const maxTimeLimitMs = this.processMaxAgeAssertionTime(
-              this.options.maxAssertionAgeMs,
-              subjectNotOnOrAfter,
-              assertion.$.IssueInstant
-            );
-
-            const subjErr = this.checkTimestampsValidityError(
-              nowMs,
-              subjectNotBefore,
-              subjectNotOnOrAfter,
-              maxTimeLimitMs
-            );
-            if (subjErr) {
-              throw subjErr;
+        if (subject[0].SubjectConfirmation?.length > 0) {
+          subjectConfirmation = subject[0].SubjectConfirmation?.find(
+            (_subjectConfirmation: XMLOutput) => {
+              let res = false;
+              if (_subjectConfirmation) {
+                const _confirmData =
+                  _subjectConfirmation && _subjectConfirmation.SubjectConfirmationData
+                    ? _subjectConfirmation.SubjectConfirmationData[0]
+                    : null;
+                if (_confirmData?.$) {
+                  const subjectNotBefore = _confirmData.$.NotBefore;
+                  const subjectNotOnOrAfter = _confirmData.$.NotOnOrAfter;
+                  const maxTimeLimitMs = this.processMaxAgeAssertionTime(
+                    this.options.maxAssertionAgeMs,
+                    subjectNotOnOrAfter,
+                    assertion.$.IssueInstant
+                  );
+
+                  const subjErr = this.checkTimestampsValidityError(
+                    nowMs,
+                    subjectNotBefore,
+                    subjectNotOnOrAfter,
+                    maxTimeLimitMs
+                  );
+                  if (subjErr === null) res = true;
+                }
+              }
+              return res;
             }
+          );
+
+          if (subjectConfirmation) {
```

Comment:
I was doing some more testing with this code and found that your previous way of doing this was just wrong. `Array.prototype.find()` doesn't return an array, so trying to index into `subjectConfirmation` with a `[0]` always returned `null`, which, always caused the following code to fail.

I also found that we don't have coverage for the following code, but that should be fixed in another PR.

Reply:
I've added code coverage reporting so you can see how your code is doing.